### ISSUE TITLE
RDL cannot generate resrc for a large node-count spec

### ISSUE THREAD

**Issue by dongahn** [flux-framework/flux-sched#199]
I am trying to do some erformance studies in preparation for #193 and discovered that RDL fails with the following msg:

```
rdl_load: Error! ...space/Flux/DB-test/flux-sched/rdl/RDL/lib/ListOf.lua:33: too many results to unpack
Error: resrc_generate_rdl_resources
```

The smallest node count that hit this for me is ~33000. A conf file:

```
uses "Node"

Hierarchy "default" {
    Resource{ "cluster", name = "hype",
    children = { ListOf{ Node,
                  ids = "0-33000",
                  args = { basename = "hype",
                           sockets = {"0-7", "8-15"},
                           memory_per_socket = 15000 }
                 },
               }
    }
}
```

**Comment by grondo** [comment]
uhoh. The RDL implementation makes an individual resource form each Node, each Socket within each node, and each Core within each socket, we may be overflowing the Lua stack by using `unpack`.

I may be able to change to iterate over whatever table is using `unpack`, but I'll have to peek at the code to remember...

**Comment by dongahn** [comment]
@grondo: Thanks for the explanation. Just to be clear, I don't need this fix right away as I already have a reasonable sized csv file for my current DB testing. (Don't suffer your Friday night over this :-)

**Comment by lipari** [comment]
@grondo, while the RDL Node class is open for surgery, would you mind modifying it to not require or instantiate a Socket by default?  I utilize the Node class with the power flow graph hierarchy and having socket and core children automatically populated is irrelevant to a node's power draw - unless it's specifically configured that way.

**Comment by grondo** [comment]
Sure, I will take a look. Mind opening a separate issue on it if you have the chance?

**Comment by lipari** [comment]
Done:  [Issue 200](https://github.com/flux-framework/flux-sched/issues/200)

**Comment by grondo** [comment]
closed by #440
